### PR TITLE
refactor(chore): fix sonar code smells

### PR DESCRIPTION
**Pull request by Surbhi-sharma1**
GH-1052

## Description

fix sonar code smells

Fixes # (issue)
GH-1052
## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Intermediate change (work in progress)

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

## Checklist:

- [ ] Performed a self-review of my own code
- [x] npm test passes on your machine
- [ ] New tests added or existing tests modified to cover all changes
- [ ] Code conforms with the style guide
- [ ] API Documentation in code was updated
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: packages/custom-sf-changelog/writer-opts.js [hunk 1]
```diff
@@ -65,7 +65,7 @@ function getWriterOpts() {
         if (url) {
           url = `${url}/issues/`
           // Issue URLs.
-          commit.subject = commit.subject.replace(/#([0-9]+)/g, (_, issue) => {
+          commit.subject = commit.subject.replace(/#(\d+)/g,(_, issue) => {
```
Comment (yeshamavani) [inline packages/custom-sf-changelog/writer-opts.js:68]: why this change

File: sandbox/auth-ms-basic-example/src/services/user-ops.service.ts [hunk 2]
```diff
@@ -116,9 +116,7 @@ export class UserOpsService {
 
   validateUserCreation(userData: UserDto, options?: AnyObject) {
     // Check for valid email
-    const emailRegex = new RegExp(
-      "[a-z0-9!#$%&'*+/=?^_`{|}~-]+(?:.[a-z0-9!#$%&'*+/=?^_`{|}~-]+)*@(?:[a-z0-9](?:[a-z0-9-]*[a-z0-9])?.)+[a-z0-9](?:[a-z0-9-]*[a-z0-9])?",
-    );
+    const emailRegex = /[a-z0-9!#$%&'*+/=?^_`{|}~-]+(?:.[a-z0-9!#$%&'*+/=?^_`{|}~-]+)*@(?:[a-z0-9](?:[a-z0-9-]*[a-z0-9])?.)+[a-z0-9](?:[a-z0-9-]*[a-z0-9])?/;
```
Comment (yeshamavani) [inline sandbox/auth-ms-basic-example/src/services/user-ops.service.ts:119]: new RegExp ?

**Comment by yeshamavani** [comment]
sonar issue in this PR

File: packages/custom-sf-changelog/writer-opts.js (hunk 1 above)
Comment (Surbhi-sharma1) [reply packages/custom-sf-changelog/writer-opts.js]: changes done

File: sandbox/auth-ms-basic-example/src/services/user-ops.service.ts (hunk 2 above)
Comment (Surbhi-sharma1) [reply sandbox/auth-ms-basic-example/src/services/user-ops.service.ts]: changes done
